src/GraphQL.Tests/Subscription/ObservableExtensionsTests.cs

```diff
@@ -0,0 +1,317 @@
+        var observable = Source
+            .SelectCatchAsync(
+                async (data, token) =>
+                {
+                    var s = int.Parse(data);
+                    await Task.Delay(s);
+                    return data;
+                },
+                (error, token) => throw error);
+        observable.Subscribe(Observer);
+        Source.Next("10");
+        await Task.Delay(500);
+        Source.Next("20");
+        await Observer.WaitForAsync("Next '10'. Next '20'. ");
+    }
+
+    [Fact]
+    public void SendSynchronously()
+    {
+        // validates that if the transformations execute synchronously,
+        // nothing gets scheduled on the task scheduler
+        var observable = Source
+            .SelectCatchAsync(
+                (data, token) => new ValueTask<string>(data),
+                (error, token) => error is ApplicationException ? throw error : new ValueTask<Exception>(new DivideByZeroException()));
+        observable.Subscribe(Observer);
+        Source.Next("a");
+        Source.Error(new ApplicationException());
+        Source.Next("b");
+        Source.Error(new InvalidTimeZoneException());
+        Source.Completed();
+        Observer.Current.ShouldBe("Next 'a'. Error 'ApplicationException'. Next 'b'. Error 'DivideByZeroException'. Completed. ");
+    }
+
+    [Fact]
+    public async Task CanceledSubscriptionsDontSendData()
+    {
+        var observable = Source
+            .SelectCatchAsync(
+                (data, token) => new ValueTask<string>(data),
+                (error, token) => error is ApplicationException ? throw error : new ValueTask<Exception>(new DivideByZeroException()));
+        var disposer = observable.Subscribe(Observer);
+        Source.Next("test");
+        Observer.Current.ShouldBe("Next 'test'. ");
+        disposer.Dispose();
+        Source.Next("a");
+        Source.Error(new ApplicationException());
+        Source.Next("b");
+        Source.Error(new InvalidTimeZoneException());
+        Source.Next("c");
+        Source.Completed();
+        await Task.Delay(200); // just in case, but should execute synchronously anyway
+        Observer.Current.ShouldBe("Next 'test'. ");
+    }
+
+    [Fact]
+    public async Task CanceledSubscriptionsDontTransform()
+    {
+        bool transformed = false;
+        bool disposed = false;
+        var observable = Source
+            .SelectCatchAsync(
+                async (data, token) =>
+                {
+                    transformed = disposed;
+                    return data;
+                },
+                async (error, token) =>
+                {
+                    transformed = disposed;
+                    return error;
+                });
+        var disposer = observable.Subscribe(Observer);
+        Source.Next("test");
+        Source.Error(new DivideByZeroException());
+        Observer.Current.ShouldBe("Next 'test'. Error 'DivideByZeroException'. ");
+        disposer.Dispose();
```

Comment:
```suggestion
        var subscription = observable.Subscribe(Observer);
        Source.Next("test");
        Source.Error(new DivideByZeroException());
        Observer.Current.ShouldBe("Next 'test'. Error 'DivideByZeroException'. ");
        subscription.Dispose();
```

Reply:
And in all other places.

Reply:
If you find any others, lmk.  I think I saw one other.